presto-main/src/main/java/com/facebook/presto/operator/scalar/IpPrefixFunctions.java

```diff
@@ -197,6 +197,54 @@ public static Block collapseIpPrefixes(@SqlType("array(IPPREFIX)") Block unsorte
         return blockBuilder.build();
     }
 
+    @Description("Split the input prefix into subnets the size of the new prefix length.")
+    @ScalarFunction("ip_prefix_subnets")
+    @SqlType("array(IPPREFIX)")
+    public static Block ipPrefixSubnets(@SqlType(StandardTypes.IPPREFIX) Slice prefix, @SqlType(StandardTypes.BIGINT) long newPrefixLength)
+    {
+        boolean inputIsIpV4 = isIpv4(prefix);
+
+        if (newPrefixLength < 0 || (inputIsIpV4 && newPrefixLength > 32) || (!inputIsIpV4 && newPrefixLength > 128)) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Invalid prefix length for IPv" + (inputIsIpV4 ? "4" : "6") + ": " + newPrefixLength);
+        }
+
+        int inputPrefixLength = getPrefixLength(prefix);
+        // We can only disaggregate a prefix if the prefix length is longer (more-specific) than the prefix length of the input prefix.
+        // E.g., if the input prefix is /24, the subnetSize must be like /25, /26 etc....
+        if (inputPrefixLength > newPrefixLength) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "newPrefixLength argument '" + newPrefixLength + "' must be the same or longer (more-specific) than the input prefix's length '" + inputPrefixLength + "'");
+        }
+
+        // Count how many new prefixes we will have.  /24 => /26s; 26 - 24 = 2; 2^2 = 4 new prefixes.
```

Comment:
Could u please use plain English instead of just some equations? We want to make the code and comment as straightforward as possible.

Reply:
Took a another pass through these comments based on your feedback.